codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/alltypesunionstructure.java

```diff
@@ -0,0 +1,3951 @@
+            return allTypesUnionStructure(AllTypesUnionStructure.builder().applyMutation(allTypesUnionStructure).build());
+        }
+    }
+
+    static final class BuilderImpl implements Builder {
+        private String stringMember;
+
+        private Integer integerMember;
+
+        private Boolean booleanMember;
+
+        private Float floatMember;
+
+        private Double doubleMember;
+
+        private Long longMember;
+
+        private Short shortMember;
+
+        private List<String> simpleList = DefaultSdkAutoConstructList.getInstance();
+
+        private List<String> listOfEnums = DefaultSdkAutoConstructList.getInstance();
+
+        private List<Map<String, String>> listOfMaps = DefaultSdkAutoConstructList.getInstance();
+
+        private List<SimpleStruct> listOfStructs = DefaultSdkAutoConstructList.getInstance();
+
+        private List<Map<String, String>> listOfMapOfEnumToString = DefaultSdkAutoConstructList.getInstance();
+
+        private List<Map<String, SimpleStruct>> listOfMapOfStringToStruct = DefaultSdkAutoConstructList.getInstance();
+
+        private Map<String, List<Integer>> mapOfStringToIntegerList = DefaultSdkAutoConstructMap.getInstance();
+
+        private Map<String, String> mapOfStringToString = DefaultSdkAutoConstructMap.getInstance();
+
+        private Map<String, SimpleStruct> mapOfStringToSimpleStruct = DefaultSdkAutoConstructMap.getInstance();
+
+        private Map<String, String> mapOfEnumToEnum = DefaultSdkAutoConstructMap.getInstance();
+
+        private Map<String, String> mapOfEnumToString = DefaultSdkAutoConstructMap.getInstance();
+
+        private Map<String, String> mapOfStringToEnum = DefaultSdkAutoConstructMap.getInstance();
+
+        private Map<String, SimpleStruct> mapOfEnumToSimpleStruct = DefaultSdkAutoConstructMap.getInstance();
+
+        private Map<String, List<String>> mapOfEnumToListOfEnums = DefaultSdkAutoConstructMap.getInstance();
+
+        private Map<String, Map<String, String>> mapOfEnumToMapOfStringToEnum = DefaultSdkAutoConstructMap.getInstance();
+
+        private Instant timestampMember;
+
+        private StructWithTimestamp structWithNestedTimestampMember;
+
+        private SdkBytes blobArg;
+
+        private StructWithNestedBlobType structWithNestedBlob;
+
+        private Map<String, SdkBytes> blobMap = DefaultSdkAutoConstructMap.getInstance();
+
+        private List<SdkBytes> listOfBlobs = DefaultSdkAutoConstructList.getInstance();
+
+        private RecursiveStructType recursiveStruct;
+
+        private BaseType polymorphicTypeWithSubTypes;
+
+        private SubTypeOne polymorphicTypeWithoutSubTypes;
+
+        private String enumType;
+
+        private Underscore_Name_Type underscore_Name_Type;
+
+        private Document myDocument;
+
+        private AllTypesUnionStructure allTypesUnionStructure;
+
+        private Type type = Type.UNKNOWN_TO_SDK_VERSION;
+
+        private Set<Type> setTypes = EnumSet.noneOf(Type.class);
```

Comment:
To support backwards-compatibility we'd need to support multiple fields being populated, and the SEP specifies that making an existing structure into a union is backwards-compatible.

It's true that things could be simplified if we made switching between union/structures backwards incompatible, because we could guarantee only one field is populated.

Reply:
I see! I thought adding `union` to an existing structure would *not* be backwards compatible so thought this would a safe change.

Reply:
Digging into the exact wording, I suppose it's less strict than that:

> When adding tagged unions to existing SDKs, maintaining backward compatibility comes first. When existing SDKs start to see structure shapes marked with the union trait, they MUST continue to expose the shape as if it is a normal structure; however, if possible, these SDKs SHOULD also expose the shape through a tagged union abstraction. New SDKs or SDKs that are building a new major version do not have such limitations and are not required to expose tagged unions as a kind of subtype of a structure.